tsl/src/compression/api.c

```diff
@@ -1398,6 +1398,7 @@ tsl_recompress_chunk_segmentwise(PG_FUNCTION_ARGS)
 	index_endscan(index_scan);
 	UnregisterSnapshot(snapshot);
 	index_close(index_rel, AccessExclusiveLock);
+	row_decompressor_close(&decompressor);
```

Comment:
Nit: we have `row_compressor_finish` and now `row_decompressor_close` which is kinda out of sync. 

Also, maybe we could put more of the decompressor finish work in the function like freeing the estate, bistate, closing indexes and anything else necessary. If not in this PR, just add a TODO on the function.

Reply:
> Nit: we have row_compressor_finish and now row_decompressor_close which is kinda out of sync.

I think "finish" might give a wrong impression that it finishes compression, so I renamed it to "close" as well.

> Also, maybe we could put more of the decompressor finish work in the function like freeing the estate, bistate, closing indexes and anything else necessary. If not in this PR, just add a TODO on the function.

Makes sense, somehow I didn't notice them. I moved them into the "close" function as well.